### ISSUE TITLE
Browser code changes have a delay of about 45-50 seconds when changing code content  in a docker-compose nginx + php-fpm proxy.

### ISSUE THREAD

**Issue by alexandrupetrini** [bitnami/bitnami-docker-php-fpm#140]
**Description**

I am having a docker-compose nginx + php-fpm as reverse proxy configuration for php development showcased [here](https://github.com/alexandrupetrini/docker-symfony-nginx-phpfpm).

I am using it for symfony and laravel.
   
Both have the same issue: about 45 to 50s delay before code changes are reflected in the browser. 

**Steps to reproduce the issue:**

1. `git clone https://github.com/alexandrupetrini/docker-symfony-nginx-phpfpm`

2. `cd docker-symfony-nginx-phpfpm`

3. `docker-compose up -d --build`

4. Visit  [http://localhost](http://localhost/) or 
[http://myapp.example.com/](http://myapp.example.com/) (have `127.0.0.1	myapp.example.com example.com www.example.com` in `/etc/hosts`)

5. In `my_app/src/Controller/MyController.php` change

`return new Response("test ")` to `return new Response("test  12345")`,

then save.

6. Refresh browser while keeping a timer running. 
 
**Describe the results you received:**

Changes appear after about 45 seconds

**Describe the results you expected:**

Changes should be almost instantaneous. 

I've lost a lot of time just to isolate it. Please help me on this one!

**Comment by miguelaeh** [comment]
Hi @alexandrupetrini ,
It seems that you are customizing our images and also adding your own application.
We have not the resources to provide this kind of personal support. It doesn't seem to be an issue with our PHP image.
Also, we are not experts in symphony, but I recommend you to check if there is any kind of cache or something that is introducing a delay in your changes.
Regards.

**Comment by alexandrupetrini** [comment]
Hi @miguelaeh , 

Thanks for your answer.
 
I've tested further the cache hypothesis. I reduced the project to a a simple `/app/my_app/public` directory with an `index.php` file in it with an `echo "test ";` line. Also nginx mounts only a vhost file with HTTP  support. 
I documented myself from [Pitfalls and Common Mistakes](https://www.nginx.com/resources/wiki/start/topics/tutorials/config_pitfalls/#config-changes-not-reflected) and other sources
But whatever I do, I cannot shake the delay. it's no longer about symfony or laravel and  not even about browser cache because curl shows the same behaviour.
I uploaded the new simplified project on the [git repo](https://github.com/alexandrupetrini/docker-symfony-nginx-phpfpm).

**Comment by alexandrupetrini** [comment]
Hi,

It seems the issue comes from bitnami/php-fpm:latest. I tried replacing it with php:7.4-fpm-alpine and the delay disappears. I also tried to replace bitnami/nginx:latest with nginx:alpine and keep bitnami/php-fpm:latest and this does not remove the delay. I think there is some setting in the bitnami fpm image that makes it happen.

**Comment by miguelaeh** [comment]
Hi @alexandrupetrini ,
I tested using your repository, and in my case, it seems to be the browser cache. I am using Firefox, and the changes are instant just reloading with `ctrl + shift + R` to ignore cache.
Could you give a try?

**Comment by alexandrupetrini** [comment]
Hello @miguelaeh,

Should there be another place for caching this? I mean I even have problems with curl which does not see the code change instantaneously. I tried firefox, chrome, chromium normal and incognito, with Disable cache checked in the Dev Tools Network tab. I am using ubuntu 20.04. My friend has the same issue on hist ubuntu work laptop and on his windows laptop with docker.

And besides that, we were using bitnami/nginx built images with Nginx, php and npm at the previous workplace and from time to time we were having this kind of issues. We were putting dd() in laravel controllers (dump() and die())  and the change just couldn't get in the browser. Eventually either by pruning docker or by pruning, purging docker from the system then reboot, reinstall docker, recreate the project would fix it. I am not the only one that had this issue at the previous workplace.

It might seem a fantasy, but this actually happened, and we were all having Ubuntu on machines.
Regards.

**Comment by alexandrupetrini** [comment]
PS: I usually use this command to start docker-compose : `docker-compose up -d --force-recreate`

**Comment by alexandrupetrini** [comment]
Hello again @miguelaeh,

I found the issue after trying to migrate the config to almost full alpine. The problem reappeared when I installed opcache extension and added an opcache.ini with the performance symfony settings. 

The problem in bitnami/php-fpm:<tag> is the `opcache.revalidate_freq=60` (seconds)

![2020-09-16_21-43](https://user-images.githubusercontent.com/44755709/93379438-816d5480-f866-11ea-8da8-57ee11aa932e.png)
![image](https://user-images.githubusercontent.com/44755709/93379904-2ee06800-f867-11ea-9c65-caea22d3f0ef.png)

To be able to have code changes properly reflected in development we should have something like this in php.ini:

`opcache.revalidate_freq=2`
`opcache.validate_timestamps=1`

The second option is enabled by default.
![2020-09-16_21-57](https://user-images.githubusercontent.com/44755709/93380327-c5ad2480-f867-11ea-9cb3-cb64d7ce156e.png)

I think this should get mentioned somewhere, as it is an easy fix and a pain to find 😁😁

Best wishes,
Alex
